### PR TITLE
Rewrite transformers gpt2 with optimizations

### PR DESCRIPTION
**Pull request by Akababa**
- Optimize the way the attention masking is done
- Remove all unused code paths
- Remove batch dimension (model files don't change)
- Small changes to gpt2generator.py for readability

Tested with the 16bit pytorch model as well as pretrained transformers models.
More detailed speed and output comparisons are welcome.
When I ran the raw inference in colab I got the exact same output when using old gelu and a varying (30%-50%) speed increase depending on the input length. The new torch.nn.GELU module from torch 1.4.0 changes the output slightly, but I haven't seen how much it affects the generated text.

### CODE REVIEW

**Comment by MikkoMMM** [comment]
I don't think I can say much except that it seems to be working. I did notice that the requirements for Torch are now much more strict. It would seem that for Python 3.7 (which I use to run Clover) I have Torch 1.3.1 installed. Is there any specific reason why we shouldn't allow anything older than 1.4.0?

**Comment by Akababa** [comment]
torch 1.4.0 has the built-in GELU activation support which is faster than the one transformers uses, although I haven't tested by how much. It should still work with the older versions of torch though so if there are some older GPUs which don't support 1.4.0 then we could make it an optional requirement instead. Can you try `pip install torch==1.4.0` again to see if you can run it?

Did you notice any differences in output or speed between this branch and master?

**Comment by MikkoMMM** [comment]
I tried with Torch 1.4.0 too and I can run it. I haven't tried to time how long generating output takes.
